src/movies.js

```diff
@@ -1,25 +1,175 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
 
+function getAllDirectors(moviesArray) {
+    let allDirectors=moviesArray.map(x=>x.director)
+   return allDirectors
+}
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+  
+
+function howManyMovies(moviesArray) {
+  let result = 0
+  if (moviesArray.length===0){
+    return result
+  }
+
+  let moviesSpielberg = moviesArray.filter (movie => movie.director===`Steven Spielberg`)
+  let onlyDrama =[]
+  for ( let i = 0; i < moviesSpielberg.length; i ++){
+    let genre = moviesSpielberg[i].genre
+
+    for ( let j = 0; j < genre.length; j++ ) {
+      const element = genre[j];
+      
+      if (element==="Drama") {
+        result++
+        onlyDrama.push(moviesSpielberg[i]) 
+      }
+      
+    }
+    
+  }
+  
+
+ return result
+
+}
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+
+function scoresAverage(moviesArray) {
+  let result = 0
+  if (moviesArray.length === 0) {//if is empty = 0
+    return result
+  }
+
+ let scores= moviesArray.map(scores=> scores.score) 
+ let suma = 0
+ for (let i = 0 ; i < scores.length; i ++){
+  let element = scores[i];
+  if (element === " " || element === "" || element === undefined ) {//if if one of the movies does not have score = 0
+    element = 0
+    
+  }
+  suma += element 
+  
+ }
+
+let avg = suma / scores.length// average 
+result = avg
+
+ return parseFloat(result.toFixed(2))//number with two decimals
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+console.log("hola")
+
+
+function dramaMoviesScore(moviesArray) {
+  
+  const dramaMovies = moviesArray.filter(movie => movie.genre.includes('Drama'));console.log(typeof dramaMovies)//creamos un array con las pelis de drama
+  if (dramaMovies.length === 0) {
+    return 0
+  }
+  let average=scoresAverage(dramaMovies)
+  return average
```

Comment:
Esto te lo cambio, primero tendria que ser const!!! pero poco a poco queremos que hagais esto otro!

```suggestion
  return scoresAverage(dramaMovies);
```